src/services/cw721/cw721.service.ts

```diff
@@ -511,4 +528,74 @@ export default class Cw721HandlerService extends BullableService {
       .join('smart_contract', 'cw721_contract.contract_id', 'smart_contract.id')
       .groupBy('cw721_contract.id');
   }
+
+  @QueueHandler({
+    queueName: BULL_JOB_NAME.REINDEX_CW721_HISTORY,
+    jobName: BULL_JOB_NAME.REINDEX_CW721_HISTORY,
+  })
+  public async crawlMissingContractHistory(
+    _payload: ICw721ReindexingHistoryParams
+  ) {
+    const {
+      smartContractId,
+      startBlock,
+      endBlock,
+      currentId,
+      contractAddress,
+    } = _payload;
+    // insert data from event_attribute_backup to event_attribute
+    const { limitRecordGet } = config.cw721.crawlMissingContract;
+    // eslint-disable-next-line no-await-in-loop
+    const events = await this.getCw721ContractEvents(
+      startBlock,
+      endBlock,
+      smartContractId,
+      { limit: limitRecordGet, currentId }
+    );
+    if (events.length > 0) {
+      await this.handleCW721Activity(events);
+      await this.createJob(
+        BULL_JOB_NAME.REINDEX_CW721_HISTORY,
+        BULL_JOB_NAME.REINDEX_CW721_HISTORY,
+        {
+          smartContractId,
+          startBlock,
+          endBlock,
+          currentId: events[events.length - 1].smart_contract_event_id,
+          contractAddress,
+        } satisfies ICw721ReindexingHistoryParams,
+        {
+          removeOnComplete: true,
+        }
+      );
+    } else {
+      const queue: Queue = this.getQueueManager().getQueue(
+        BULL_JOB_NAME.REINDEX_CW721_CONTRACT
+      );
+      (await queue.getJob(contractAddress))?.remove();
```

Comment:
ko có thì thôi sao phải remove ?

Reply:
em thử cách update 1 job chưa? nó có khác ko?

Reply:
![image](https://github.com/aura-nw/horoscope-v2/assets/49814372/0f351b3e-adef-4896-ad1f-8b836293fe9f)

Reply:
Em có làm để nó block nếu user đáp 2 job cùng contract address vào Queue 1, sau khi done toàn bộ luồng mới remove job đó trên Queue 1 để dừng blocking ạ